cmd/csaf_provider/actions.go

```diff
@@ -195,9 +195,10 @@ func (c *controller) upload(r *http.Request) (any, error) {
 	var dynamicCategories []string
 	if catExprs := c.cfg.DynamicCategories(); len(catExprs) > 0 {
 		var err error
-		if dynamicCategories, err = pe.Strings(catExprs, true, content); err != nil {
+		if dynamicCategories, err = pe.StringsFromTree(
+			catExprs, true, content); err != nil {
```

Comment:
@s-l-teichmann As discussed: I checked with `false`: This fails also when the JSONPath is not in the document...
```
eval of dynamic category expressions failed: extract failed 'document.lang': unsupported type: <nil>
```

As this is not the desired output, we either need to manage the `nil` differently or keep the `true`. At the moment, I'm in favor of the last option. If you prefer that too, please resole the conversations and add an issue about splitting these two cases (expr valid but not in doc vs. expr invalid).

Reply:
I am going to fix this.

Reply:
With 0745a0943d4050e864f00f9fc83a075d32d51d60 the compilation of the expressions are separated from the execution. Errors produced by compiling are reported, errors from evaluations are ignored.
That should address this.